### PR TITLE
You can no longer get mauled by 20 rabid corgis in a duffel bag 

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
In my dog AI PR, I forgot to add a check for whether a dog in harass mode was actually on a turf when harassing, meaning that an angry pup would still be capably of biting from inside a locker or backpack. I *also* found out you can fit dogs in backpacks. Who knew??

So dogs now need to actually be on a normal turf to harass/attack, with the exception of void corgis since maintainers thought it was funny (to which I agree!) So if you see someone stuffing a bunch of void corgis into a backpack, you should probably make sure you're on their good side.

## Why It's Good For The Game
Reduces jank

## Changelog
:cl: Ryll/Shaps
fix: Dogs are no longer able to maul you from inside backpacks or lockers. Unless they're void corgis, those pups are still more than capable of doing so!
/:cl:

### CODE REVIEW

**Comment by TheBonded** [comment]
So I can't hold a corgi in my hand and use it to bite people? Sad!

**Comment by TheBonded** [comment]
But in all honesty, I remember when I found this glitch and tested it out with a BoH, could deepcrit in one tick, so it is a good thing it's being fixed. I do think they should be able to attack when held, but it's not the end of the world if they're not

It is however, the end of the world if we can't wear corgis as hats soon. Seriously, someone with spriting experience get on this!

**Comment by Serxule** [comment]
i mean, this is possible irl, maybe just limit the number of dogs in a bag to 3 (accounts for all head dogs)

**Comment by TheBonded** [comment]
> i mean, this is possible irl, maybe just limit the number of dogs in a bag to 3 (accounts for all head dogs)

That doesn't work, since dogs are normal sized. Maybe let bags hold large sized items, 2 for satchel or 3 for duffel bag, takes up the space of 3 normal items. Then you can hold any large item in your bag, would be a good change

File: code/datums/ai/dog/dog_behaviors.dm [hunk 1]
```diff
@@ -168,7 +168,7 @@
 /datum/ai_behavior/harass/perform(delta_time, datum/ai_controller/controller)
 	. = ..()
 	var/mob/living/living_pawn = controller.pawn
-	if(!istype(living_pawn))
+	if(!istype(living_pawn) || (!isturf(living_pawn.loc) || HAS_TRAIT(living_pawn, TRAIT_AI_BAGATTACK)))
```
Comment (Ghommie) [inline code/datums/ai/dog/dog_behaviors.dm:171]:
```suggestion
	if(!istype(living_pawn) || !(isturf(living_pawn.loc) || HAS_TRAIT(living_pawn, TRAIT_AI_BAGATTACK)))
```

**Comment by ATH1909** [comment]
image

File: code/datums/ai/dog/dog_behaviors.dm [hunk 2]
```diff
@@ -168,7 +168,7 @@
 /datum/ai_behavior/harass/perform(delta_time, datum/ai_controller/controller)
 	. = ..()
 	var/mob/living/living_pawn = controller.pawn
-	if(!istype(living_pawn))
+	if(!istype(living_pawn) || !(isturf(living_pawn.loc) || HAS_TRAIT(living_pawn, TRAIT_AI_BAGATTACK)))
```
Comment (ATH1909) [inline code/datums/ai/dog/dog_behaviors.dm:171]: you could probably add some logic similar to (but not quite the same as) that of stabilized extracts to let dogs that are held in someone's hand (or pocket) attack people, but not dogs that are inside of an object or something. something like a check to see if the dog is inside of a mob holder, and then an isliving(living_pawn.loc.loc) check or something

**Comment by ATH1909** [comment]
also, top tier branch name

**Comment by optimumtact** [comment]
remove the ability to pick up pets it's poorly implemented

**Comment by TheBonded** [comment]
> remove the ability to pick up pets it's poorly implemented

What? Why?

**Comment by LemonInTheDark** [comment]
> why

**Comment by optimumtact** [comment]
because it's poorly implemented

**Comment by TheBonded** [comment]
> because it's poorly implemented

How so? Also, isn't it better to fix issues instead of removing them entirely? Weren't you the one who told me lazy fixes like removing things was bad, and you should try to fix things if at all possible?

**Comment by optimumtact** [comment]
It's always better to fix issues by removing them

**Pull request merged**